File: src/main/java/com/gabrielczar/semantic/entities/UnMatchingEntry.java
package com.gabrielczar.semantic.entities;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.sql.Date;

@Data
@NoArgsConstructor
@Entity(name = "un_matching_entries")
public class UnMatchingEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long serial;

    private String token;

    private Integer id;

    private Date date;

    private Double latitude;

    private Double longitude;

    public UnMatchingEntry(String token, Integer id, Date date, Double latitude, Double longitude) {
        this.token = token;
        this.id = id;
        this.date = date;
        this.latitude = latitude;
        this.longitude = longitude;
    }
}

File: src/main/java/com/gabrielczar/semantic/entities/UnMatchingEntry.kt
package com.gabrielczar.semantic.entities

import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity(name = "un_matching_entries")
data class UnMatchingEntry (
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var serial : Long? = null,
        var token  : String,
        var id : Int,
        var date : Date,
        var latitude : Double,
        var longitude : Double
) {
    constructor(token: String, id: Int, date: Date, latitude: Double, longitude: Double) : this(null, token, id, date, latitude, longitude)
}
